syntax = "proto3";

package resdb;

import "proto/signature_info.proto";
import "proto/durable.proto";

message ReplicaInfo {
    int64 id = 1;
    optional string ip = 2;
    optional int32 port = 3;
    optional CertificateInfo certificate_info = 4; // include the private/public key, public key of admin and the certificate published by admin.
}

message RegionInfo {
  repeated ReplicaInfo replica_info = 1;
  int32 region_id = 2;
}

message ResConfigData{
  repeated RegionInfo region = 1;
  int32 self_region_id = 2;
  optional RocksDBInfo rocksdb_info = 3;
  optional LevelDBInfo leveldb_info = 4;
}

message ReplicaStates {
  repeated ReplicaState state = 1;
}

message ReplicaState {
  int64 view = 1;
  ReplicaInfo replica_info = 2;
}
